# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/bindings/Exceptions.h
# Commit: 65cc1910918c
# Full Hash: 65cc1910918cf410ada326fd353db1da51c2c0eb
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-12-07 21:55:34
# Regressor Bug: 1588839
# File Overlap Count: 1
# Description:
#   Bug 1588839 - Part 1 - Add helper to allow structured cloning Error/Exception objects. r=bzbarsky
#   
#   In order to be able to reasonably debug error results from things like
#   JSWindowActor.sendQuery, we need to be able to clone errors across process
#   boundaries, so that they can be propagated to the caller that initiated a
# ==============================================================================

diff -r 2f8b5b48c896 -r 65cc1910918c dom/bindings/Exceptions.h
--- a/dom/bindings/Exceptions.h	Mon Dec 02 18:28:28 2019 +0000
+++ b/dom/bindings/Exceptions.h	Sat Dec 07 04:43:24 2019 +0000
@@ -55,6 +55,11 @@
 already_AddRefed<nsIStackFrame> CreateStack(JSContext* aCx,
                                             JS::StackCapture&& aCaptureMode);
 
+// Like the above, but creates a JSStackFrame wrapper for an existing
+// JS::SavedFrame object, passed as aStack.
+already_AddRefed<nsIStackFrame> CreateStack(JSContext* aCx,
+                                            JS::Handle<JSObject*> aStack);
+
 }  // namespace exceptions
 }  // namespace dom
 }  // namespace mozilla